tools/confusion_matrix.py

```diff
@@ -0,0 +1,175 @@
+import argparse
+import os
+
+import matplotlib.pyplot as plt
+import mmcv
+import numpy as np
+from matplotlib.ticker import MultipleLocator
+from mmcv import Config, DictAction
+
+from mmseg.datasets import build_dataset
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Generate confusion matrix from segmentation results')
+    parser.add_argument('config', help='test config file path')
+    parser.add_argument(
+        'prediction_path', help='prediction path where test .pkl result')
```

Comment:
Currently we do not generate .pkl result like MMDetection, neither do prediction result. ;(

Reply:
Oh. I thought ```test.py``` generates either prediction result or eval result depending on args.eval argument. It .pkl format doesn't work, then ```confusion_matrix.py``` can take just the numpy type of prediction result from 
```python 
        results = multi_gpu_test(
            model,
            data_loader,
            args.tmpdir,
            args.gpu_collect,
            False,
            pre_eval=args.eval is not None and not eval_on_format_results,
            format_only=args.format_only or eval_on_format_results,
            format_args=eval_kwargs)
```
instead and it would work the same.

Reply:
OK, let me try to test whether `.pkl` could work, actually we never used pickle format before.

Reply:
Sure! The result(numpy array) can be stored in any kind of format tho :)